From aa1fb8dc51767f10b52f0eae2ecaf180865fbbe0 Mon Sep 17 00:00:00 2001
From: Lorenzo Pirritano <lpirritano@anaconda.com>
Date: Fri, 20 Dec 2024 18:20:45 +0100
Subject: [PATCH 2/2] set visibility preset for absl

---
 CMakeLists.txt | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93bb8d2..56666a7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,6 +59,21 @@ if(UNIX)
   find_package(Threads REQUIRED)
 endif()
 
+# Abseil
+# In the normal course of things, we would not need the following constraint.
+# However, if we do not set the standard, then std::basic_string_view is not
+# available, which absl depends on and linking the test fails.
+set(CMAKE_CXX_STANDARD 17)
+# Abseil does not handle symbol visibility correctly. As a workaround, we let it
+# expose all the symbols to avoid linker errors.
+set(CMAKE_CXX_VISIBILITY_PRESET default)
+# ABSL plans to enable this flag to be on by default in the future. If it is not
+# set, then it issues a warning. We force enable it, both to silence the warning
+# but to also be safe going forward.
+option(ABSL_PROPAGATE_CXX_STD "Use CMake C++ standard meta features
+(e.g. cxx_std_14) that propagate to targets that link to Abseil" ON)
+option(ABSL_ENABLE_INSTALL "Enable install rules" ON)
+
 set(ABSL_DEPS
     absl_base
     absl_core_headers
@@ -82,6 +97,9 @@ if(NOT TARGET absl::base)
 endif()
 list(APPEND REQUIRES ${ABSL_DEPS})
 
+unset(CMAKE_CXX_STANDARD)
+unset(CMAKE_CXX_VISIBILITY_PRESET)
+
 if(RE2_USE_ICU)
   if(NOT TARGET ICU::uc)
     find_package(ICU REQUIRED COMPONENTS uc)
-- 
2.39.1

